test/e2e/operator/garden/common.go

```diff
@@ -116,3 +116,13 @@ func waitForGardenToBeDeleted(ctx context.Context, garden *operatorv1alpha1.Gard
 		return runtimeClient.Get(ctx, client.ObjectKeyFromObject(garden), garden)
 	}).WithPolling(2 * time.Second).Should(BeNotFoundError())
 }
+
+func cleanupVolumes(ctx context.Context) {
+	Expect(runtimeClient.DeleteAllOf(ctx, &corev1.PersistentVolumeClaim{}, client.InNamespace("garden"))).To(Succeed())
+
+	CEventually(ctx, func(g Gomega) []corev1.PersistentVolume {
+		pvList := &corev1.PersistentVolumeList{}
+		g.Expect(runtimeClient.List(ctx, pvList)).To(Succeed())
+		return pvList.Items
+	}).WithPolling(2 * time.Second).Should(HaveLen(0))
```

Comment:
This might work with the existing test setup but will cause problems if the e2e tests run against an existing cluster. I think we should rather selectively check the if there are no `PV`s referring to `PVC`s in the `garden` namespace.